File: python/QuintoEsercizio.py
# 1. popolare un array di N elementi interi, quindi chiedere un numero intero e visualizzare se è presente o meno
#se è presente indicare quante volte è presente e le posizioni in cui è presente

#analisi dei requisiti

#risorse: array di N elementi interi, numeri interi

#richiesta: visualizzare se il numero inserito é presente o meno

n=0
# 1. PREPARAZIONE: chiediamo la dimensione N
while True:
    n = int(input("quanti numeri vuoi inserire?"))
    if n>0:
        break
    print("per favore, inserisci un numero maggiore di 0: ")

# 2. POPOLAMENTO: riempiamo la lista
numeri = []
for i in range (n):
    valore = int (input("inserisci un numero: "))
    numeri.append(valore) #aggiungiamo il numero alla lista

# 3. RICERCA: chiediamo cosa cercare e analizziamo la lista
while True:
    bersaglio = int(input("che numero vuoi cercare?"))
    conteggio = 0
    posizioni = []
    for i in range (len(numeri)):  #len(numeri) ci dice quanto é lunga la lista
        if numeri[i] == bersaglio:
            conteggio = conteggio +1
            posizioni.append(i+1)

    if conteggio == 0:
        print(f"il numero {bersaglio} non e' presente nella lista.")
    else:
        print(f"il numero {bersaglio} é presente")
        print(f"appare {conteggio} volta/e")
        print(f"si trova nelle seguenti posizioni: {posizioni}°")
        break

File: c/QuintoEsercizio.c

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int main(){
   
    //TRACCIA

// chiedere all utente N numeri al max 100 numeri interi compresi tra 0 e 2026
// visualizzare tutti i numeri almeno duplicati
// per ogni numero almeno duplicato visualizzare su msg del tipo
// "il numero 4 è presente 3 volte nelle posizioni 5 8 e 10"
// un ulteriore array alla fine 
    
     // RISORSE DA DEFINIRE
 // N                conterrà QUANTI
 // numeri           array di N elementi che contiene i numeri interi (min 2 max 100)
 // conta            quanti duplicati ci sono
 // duplicati     array (max 100) che contiene True se quella posizione contiene un duplicato 
 //                                     False se quella posizione NON contiene un duplicato
 // posizioni        array (max 100) che contiene le posizioni dei valori duplicati

    //STEP DA SEGUIRE

//1. chiedere in input QUANTI numeri vuole inserire (da 0 a 2026)
//2. inserire i QUANTI numeri nell array
//3. popolare l array NUMERI (da 0 a 2026)
//4. per ogni iterazione dobbiamo
    //a. verificare se ci sono duplicati
    //    se ci sono duplicati bisogna contare i duplicati nella variabile CONTA
    //    memorizzare le posizioni in un nuovo array POSIZIONI
    //    contrassegnare quella posizione con TRUE
//5. Per ogni duplicato stampare 
    
    int N = 0;
    const int DIMENSIONE_MAX = 100;
    const int DIMENSIONE_MIN = 2;
    int numeri[DIMENSIONE_MAX];
    for(int i=0;i<DIMENSIONE_MAX;i++){
             numeri[i] = 0;
            }
    int conta = 0;

    bool duplicati[DIMENSIONE_MAX];
    for(int i=0;i<DIMENSIONE_MAX;i++){
             duplicati[i] = false;
            }
    
    int posizioni[DIMENSIONE_MAX];
     for(int i=0;i<DIMENSIONE_MAX;i++){
             posizioni[i] = 0;
            }  


     do { 
            printf("Quanti numeri vuoi inserire? (massimo 100)");
            scanf("%d",&N);       
        } while (N < DIMENSIONE_MIN || N > DIMENSIONE_MAX);
         

        for (int i = 0;i < N;i++) {
            do
            {
                printf("Inserisci il %d* numero",i+1);
                scanf("%d", &numeri[i]);  
            } while (numeri[i]<0 || numeri[i]>2026);
              
            
        }
}